src/collective/fingerpointing/subscribers/lifecycle_logger.py

```diff
@@ -13,18 +14,23 @@ def lifecycle_logger(obj, event):
     """Log content type life cycle events like object creation,
     modification and removal.
     """
-    record = IFingerPointingSettings.__identifier__ + '.audit_lifecycle'
-    if api.portal.get_registry_record(record):
-        user, ip = get_request_information()
+    # subscriber is registered even if package has not yet been installed
+    try:
+        record = IFingerPointingSettings.__identifier__ + '.audit_lifecycle'
+        if api.portal.get_registry_record(record):
+            user, ip = get_request_information()
 
-        if IObjectCreatedEvent.providedBy(event):
-            action = 'object created'
-            extras = 'object={0}'.format(obj)
-        if IObjectModifiedEvent.providedBy(event):
-            action = 'object modified'
-            extras = 'object={0}'.format(obj)
-        if IObjectRemovedEvent.providedBy(event):
-            action = 'object removed'
-            extras = 'object={0}'.format(obj)
+            if IObjectCreatedEvent.providedBy(event):
+                action = 'object created'
+                extras = 'object={0}'.format(obj)
+            if IObjectModifiedEvent.providedBy(event):
+                action = 'object modified'
+                extras = 'object={0}'.format(obj)
+            if IObjectRemovedEvent.providedBy(event):
+                action = 'object removed'
+                extras = 'object={0}'.format(obj)
 
-        logger.info(AUDIT_MESSAGE.format(user, ip, action, extras))
+            logger.info(AUDIT_MESSAGE.format(user, ip, action, extras))
+
+    except InvalidParameterError:
+        pass  # ignore any error caused by missing registry records
```

Comment:
I think we need to keep the try / except just into the place where the exception is triggered, this way we can have a better control on where the exception happen.  Other idea is to keep less indentation, something like this:

    def iterate_logger(event):
        """Log events like content editing checkouts and checkins, if
        plone.app.iterate is installed.
        """
        # subscriber is registered even if package has not yet been installed
        record = IFingerPointingSettings.__identifier__ + '.audit_iterate'
        try:
            registries = api.portal.get_registry_record(record)
            if not registries:
                return
        except InvalidParameterError:
            return  # ignore any error caused by missing registry records
    
        user, ip = get_request_information()
        if ICheckoutEvent.providedBy(event):
            action = 'object checked out'
            extras = 'object={0} working_copy={1}'.format(
                event.object, event.working_copy)
        elif ICheckinEvent.providedBy(event):
            action = 'working copy checked in'
            extras = 'object={0} baseline={1}'.format(
                event.object, event.baseline)
        elif ICancelCheckoutEvent.providedBy(event):
            action = 'working copy cancelled'
            extras = 'object={0} baseline={1}'.format(
                event.object, event.baseline)
        logger.info(AUDIT_MESSAGE.format(user, ip, action, extras))